src/useCalendar.ts

```diff
@@ -96,7 +96,8 @@ export function useCalendar(
   const initialMonth = getInitialMonth(props, dateLib);
   const [firstMonth, setFirstMonth] = useControlledValue(
     initialMonth,
-    props.month ? startOfMonth(props.month) : undefined
+    // initialMonth is always computed from props.month if provided
+    props.month ? initialMonth : undefined
```

Comment:
Good catch! I always thought that `useControlledValue` call was hiding some bugs.

Worth to add a test to `getInitialMonth.test.ts`. Would you add one? Thanks!

Reply:
For sure, just pushed more tests 👍 
I think `getInitialMonth.test.ts` was testing well in most of the use cases with the `month` prop, but in any case, I expanded existing tests and added more.
Let me know if you think it would be useful to add tests for the `useCalendar` specifically. We may not need it because it's indirectly tested by other test cases.